### PR TITLE
Upgrade to webpack-dev-server 4

### PR DESCRIPTION
**Pull request by LukeSheard**
Upgrade our internal dev server for the webpack build experience to `webpack-dev-server@4.x` this brings with it a tonne of breaking changes to the internal configurations we needed, but outwardly there should be no breaking changes from an API point of view since it's still running `webpack@4` internally.

### CODE REVIEW

File: packages/modular-scripts/react-dev-utils/WebpackDevServerUtils.js [hunk 1]
```diff
@@ -177,7 +179,7 @@ function createCompiler({
             'Files successfully emitted, waiting for typecheck results...',
           ),
         );
-      }, 100);
+      }, 0);
```
Comment (cristiano-belloni) [inline packages/modular-scripts/react-dev-utils/WebpackDevServerUtils.js:182]: Why do we have a timeout?

File: packages/modular-scripts/react-dev-utils/logger.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+const DEBUG =
+  process.env.MODULAR_LOGGER_DEBUG || process.argv.includes('--verbose');
+const SILENT = process.env.MODULAR_LOGGER_MUTE;
+const isInteractive = process.stdout.isTTY;
+
+function printStdErr(x) {
+  if (SILENT) {
+    return;
+  }
+  x.split('\n').forEach((l) => {
```
Comment (cristiano-belloni) [inline packages/modular-scripts/react-dev-utils/logger.js:17]: Wouldn't `.split(/\r?\n/)` be more universal?

File: packages/modular-scripts/react-dev-utils/WebpackDevServerUtils.js (hunk 1 above)
Comment (LukeSheard) [reply packages/modular-scripts/react-dev-utils/WebpackDevServerUtils.js:182]: In most cases the diagnostics should come back pretty quickly - so this timeout is in place to ensure we give TypeScript sufficient time to process the files.

File: packages/modular-scripts/react-dev-utils/logger.js (hunk 2 above)
Comment (LukeSheard) [reply packages/modular-scripts/react-dev-utils/logger.js:17]: Yeah - when I wrote this initially in the main `modular-scripts` package I didn't consider it - I'll update both in place.

**Review by cristiano-belloni — APPROVED** [review]
Lgtm

**Pull request merged**